io/src/test/scala/sbt/nio/GlobSyntaxSpec.scala

```diff
@@ -0,0 +1,119 @@
+package sbt.nio
+
+import java.nio.file._
+
+import org.scalatest.FlatSpec
+import sbt.io.syntax._
+import sbt.io.AllPassFilter
+import sbt.nio.syntax._
+
+import scala.util.{ Properties, Try }
+
+class GlobSyntaxSpec extends FlatSpec {
+  import GlobSyntaxSpec._
+  "path builders" should "use `*` and `**` objects" in {
+    val path = Paths.get("/foo/bar").toAbsolutePath
```

Comment:
It makes sense for a `Glob`'s `/` to return a new `Glob`, but does it make sense for `java.nio.file.Path`?
I'd expect `base / "foo"` to return another `Path`.

Reply:
And now we're back to the reason why I wanted `**` and `*`.

Reply:
Writing a dsl that satisfies everyone is hard.

Reply:
What if base / "foo" returned `Path with Glob`? I kind of like that.

Reply:
The more I think about this, the less certain I am that Glob is defined correctly. Conceptually, a file tree glob is a sequence of filters that are applied at successive levels of the file tree. What is tricky is that Path is effectively a special case of Glob. You do make a good point that '/' shouldn't lift a Path to a Glob.  I think the proper solution is a custom file system. Since there isn't really time for that, I think the best way forward is to private[sbt] the methods in the Glob base trait and just provide the `/` operator in GlobOps. Globs could then be instantiated using the PathFinder dsl and using the stringToGlob implicit conversion function that I already added. While I don't love this solution, it buys the most time to properly solve the problem while still having at least a somewhat ergonomic api, e.g. `set watchTriggers += s"${baseDirectory.value}/**/*.scala"` is decent.

Reply:
Maybe in another PR you could treat `"/"` as path separator regardless of the operating system.

Reply:
For reference, here's better-files https://github.com/pathikrit/better-files#globbing:

```scala
val dir = "src"/"test"
val matches: Iterator[File] = dir.glob("*.{java,scala}")
```

(except their `glob` means our `globRecursive` by default)